### ISSUE TITLE
[java] A false negative caused by compound constant expression

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3454]
**Affects PMD Version:** 

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_XXX_XXX.html#XXX>

**Description:**

**Code Sample demonstrating the issue:**

```

```

**Expected outcome:**

PMD should report a violation at line ..., but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
